feature-libs/cart/wish-list/root/components/add-to-wishlist/add-to-wish-list.component.ts

```diff
@@ -17,15 +24,10 @@ export class AddToWishListComponent {
     tap((product) => this.setStockInfo(product))
   );
 
-  wishListEntries$: Observable<OrderEntry[]> = this.wishListFacade
-    .getWishList()
-    .pipe(
-      filter((wishlist) => Boolean(wishlist)),
-      map((wishList) => wishList.entries ?? [])
-    );
+  wishListEntries$: Observable<OrderEntry[]> = this.getWishListEntries();
 
   userLoggedIn$: Observable<boolean> = this.authService.isUserLoggedIn();
-  loading$: Observable<boolean> = this.wishListFacade.getWishListLoading();
+  loading$: Observable<boolean> = this.getWishListLoading();
```

Comment:
how about this:
```
wishListEntries$: Observable<OrderEntry[]>;
loading$: Observable<boolean>;

  userLoggedIn$: Observable<boolean> = this.authService.isUserLoggedIn().pipe(
    tap((isLogin) => {
      if (isLogin) {
        this.wishListEntries$ = this.wishListFacade.getWishList().pipe(
          filter((wishlist) => Boolean(wishlist)),
          map((wishList) => wishList.entries ?? [])
        );
        this.loading$ = this.wishListFacade.getWishListLoading();
      }
    })
  );
```
In html tmeplate, when `wishListEntries$` or `loading$` is accessed, the user must be logged in.

Reply:
or this:
```
  private _wishListEntries$: Observable<OrderEntry[]>;
  private _loading$: Observable<boolean>;

  get wishListEntries$(): Observable<OrderEntry[]> {
    if (this._wishListEntries$ === undefined) {
      this._wishListEntries$ = this.wishListFacade.getWishList().pipe(
        filter((wishlist) => Boolean(wishlist)),
        map((wishList) => wishList.entries ?? [])
      );
    }
    return this._wishListEntries$;
  }

  get loading$(): Observable<boolean> {
    if (this._loading$ === undefined) {
      this._loading$ = this.wishListFacade.getWishListLoading();
    }
    return this._loading$;
  }
```

Reply:
The thing is, when `getWishListEntries()` is called, we know that the user is login. So, `return of([]);` will never be executed (same as `return of(false);`). We  subscribe `this.authService.isUserLoggedIn()` in 2 places. When I read the codes, my first feeling is function `wishListDataRequired` is redundant.

Reply:
I updated it to follow your suggestion.